crates/optimism/chainspec/src/lib.rs

```diff
@@ -67,6 +68,16 @@ impl OpChainSpecBuilder {
         let forks = OP_MAINNET.hardforks.clone();
         inner = inner.with_forks(forks);
 
+        let is_bedrock_active = OP_MAINNET.is_bedrock_active_at_block(105235063);
+
+        if !is_bedrock_active {
+            warn!(
+                "Warning: Op-mainnet has been launched without importing the pre-Bedrock state. \
+                The chain will not progress without this import. \
+                Please ensure that the pre-Bedrock state is imported to avoid synchronization issues and ensure proper chain operation."
+            );
```

Comment:
this isn't quite right, because the chainspec only tracks static activation.

but we need checks this against the local database when the chain is op-mainnet

I believe the best location to do this would be somewhere here

https://github.com/paradigmxyz/reth/blob/78415ff7c586eb11097d5181153e7b7d2c092aac/crates/node/builder/src/launch/common.rs#L829-L829

we could introduce a helper function that is something like

```
fn chain_specific_db_checks
```

which would only check if the chain id is op-mainnet for example

wdyt @joshieDo

Reply:
I was thinking of putting 

`pub fn chain_specific_db_checks(&self) -> bool {
        self.chain_spec().chain_id() == 10
    }` 

inside `impl<DB, ChainSpec> LaunchContextWith<Attached<WithConfigs<ChainSpec>, DB>>`, since it has access to the chain config and database. Then I’d call that function inside `pub fn check_pipeline_consistency(&self)` to check the same. Let me know if this is a good spot for `fn chain_specific_db_checks` and if the approach makes sense!

Reply:
I was thinking of puting 

    pub fn chain_specific_db_checks(&self) -> bool {
        self.chain_spec().chain_id() == 10
    }

inside the same `impl<T, CB>LaunchContextWith<Attached<WithConfigs<<T::Types as NodeTypes>::ChainSpec>, WithComponents<T, CB>>,>`. Then call that function inside `pub fn check_pipeline_consistency(&self) -> ProviderResult<Option<B256>>` and check the same, let me know if that is a good spot to put the `fn chain_specific_db_checks` and if that if that is a good approach!